homeassistant/components/tag/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""The Tag integration."""
+import logging
+import typing
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import collection
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.storage import Store
+from homeassistant.loader import bind_hass
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+DEVICE_ID = "device_id"
+EVENT_TAG_SCANNED = "tag_scanned"
+LAST_SCANNED = "last_scanned"
+STORAGE_KEY = DOMAIN
+STORAGE_VERSION = 1
+TAG_ID = "tag_id"
+TAGS = "tags"
+
+CREATE_FIELDS = {vol.Required(TAG_ID): cv.string}
+
+UPDATE_FIELDS = {
+    vol.Required(CONF_NAME): vol.All(str, vol.Length(min=1)),
```

Comment:
In an update all fields should be optional. 
```suggestion
    vol.Optional(CONF_NAME): vol.All(str, vol.Length(min=1)),
```

Reply:
updated